org.eclipse.xtext.m2e/src/org/eclipse/xtext/m2e/XtextProjectConfigurator.java

```diff
@@ -47,7 +48,7 @@ public void configure(ProjectConfigurationRequest request,
 
 	private void configureLanguages(ProjectConfigurationRequest request, IProgressMonitor monitor) throws CoreException {
 		for (MojoExecution execution : getMojoExecutions(request, monitor)) {
-			Languages languages = getParameterValue("languages", Languages.class, request.getMavenSession(), execution);
+			Languages languages = maven.getMojoParameterValue(request.getMavenProject(), execution, "languages", Languages.class, new NullProgressMonitor());
```

Comment:
There is already an IProgressMonitor passed in here. It should be reused by using SubMonitor.convert.
Best would be to create a SubMonitor with length of getMojoExecutions and split it by 1 for each loop iteration.
Roughly:
```
List<MojoExecution> executions = getMojoExecutions(request, monitor);
SubMonitor progress = SubMonitor.convert(monitor, executions.size());
for (MojoExecution execution : executions) {
   Languages languages = maven.getMojoParameterValue(request.getMavenProject(), execution, "languages", Languages.class, progress.split(1));

Reply:
@kthoms updated the pr. ok so?